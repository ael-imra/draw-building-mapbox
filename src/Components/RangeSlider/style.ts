import { SxProps } from '@mui/material';

export const sx: { [key: string]: SxProps } = {
  rangeContainer: {
    flexFlow: 'column nowrap',
    width: '80%',
    gap: '10px',
    margin: '15px 0',
  },
  rangeTitle: {
    fontSize: 18,
    fontWeight: 500,
    color: '#555555',
  },
  sliderContainer: {
    flexFlow: 'row nowrap',
    gap: '20px',
    alignItems: 'start',
    justifyContent: 'center',
  },
  slider: {
    color: '#c5c5c4',
  },
  sliderValue: {
    minWidth: '30px',
    minHeight: '30px',
    width: '30px',
    height: '30px',
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: '#757575',
    color: 'white',
    borderRadius: '50%',
  },
};
