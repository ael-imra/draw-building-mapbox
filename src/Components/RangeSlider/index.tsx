import { Grid, Slider } from '@mui/material';
import { useState } from 'react';
import { sx } from './style';

interface RangeSliderProps {
  title: string;
  min: number;
  max: number;
  onChange?: Function;
}

const RangeSlider = (props: RangeSliderProps) => {
  const { title, min, max, onChange } = props;

  const [value, setValue] = useState<number>(min);

  const handleSlideChange = (event: Event | React.SyntheticEvent<Element, Event>, newValue: number | number[]) => {
    if (typeof newValue === 'number') {
      if (onChange) onChange(newValue);
      setValue(newValue);
    }
  };

  return (
    <Grid container sx={sx.rangeContainer}>
      <Grid sx={sx.rangeTitle}>{title}</Grid>
      <Grid container sx={sx.sliderContainer}>
        <Slider
          sx={sx.slider}
          disableSwap
          defaultValue={min}
          min={min}
          max={max}
          marks={[
            { value: min, label: min },
            { value: max, label: max },
          ]}
          valueLabelDisplay="auto"
          onChangeCommitted={handleSlideChange}
        />
        <Grid container sx={sx.sliderValue}>
          {value}
        </Grid>
      </Grid>
    </Grid>
  );
};

export default RangeSlider;
