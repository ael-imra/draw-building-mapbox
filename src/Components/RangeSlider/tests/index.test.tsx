import { render, fireEvent, screen, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import RangeSlider from '../';

afterEach(cleanup);
describe('RangeSlider', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<RangeSlider title={'title'} min={0} max={100} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render the title and slider with min and max values', () => {
    const title = 'Test Title';
    const min = 0;
    const max = 100;

    render(<RangeSlider title={title} min={min} max={max} />);

    expect(screen.getByText(title)).toBeInTheDocument();

    const slider = screen.getByRole('slider');

    expect(slider).toBeInTheDocument();
    expect(slider).toHaveAttribute('min', `${min}`);
    expect(slider).toHaveAttribute('max', `${max}`);
  });

  it('should call onChange when the slider is changed', () => {
    const title = 'Test Title';
    const min = 0;
    const max = 100;

    let value: number | undefined;

    function handleChange(val: number) {
      value = val;
    }

    render(<RangeSlider title={title} min={min} max={max} onChange={handleChange} />);

    fireEvent.change(screen.getByRole('slider'), { target: { value: 5 } });

    expect(value).toBe(5);
  });
});
