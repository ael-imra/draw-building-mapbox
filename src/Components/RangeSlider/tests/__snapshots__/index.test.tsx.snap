// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RangeSlider should render correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container css-fvth8m-MuiGrid-root"
  >
    <div
      class="MuiGrid-root css-1y7x7wc-MuiGrid-root"
    >
      title
    </div>
    <div
      class="MuiGrid-root MuiGrid-container css-1u2cepy-MuiGrid-root"
    >
      <span
        class="MuiSlider-root MuiSlider-marked MuiSlider-colorPrimary MuiSlider-sizeMedium css-1fdx4to-MuiSlider-root"
      >
        <span
          class="MuiSlider-rail css-14pt78w-MuiSlider-rail"
        />
        <span
          class="MuiSlider-track css-1gv0vcd-MuiSlider-track"
          style="left: 0%; width: 0%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="0"
          style="left: 0%;"
        />
        <span
          aria-hidden="true"
          class="MuiSlider-markLabel MuiSlider-markLabelActive css-1eoe787-MuiSlider-markLabel"
          data-index="0"
          style="left: 0%;"
        >
          0
        </span>
        <span
          class="MuiSlider-mark css-1tfve6q-MuiSlider-mark"
          data-index="1"
          style="left: 100%;"
        />
        <span
          aria-hidden="true"
          class="MuiSlider-markLabel css-yafthl-MuiSlider-markLabel"
          data-index="1"
          style="left: 100%;"
        >
          100
        </span>
        <span
          class="MuiSlider-thumb MuiSlider-thumbSizeMedium MuiSlider-thumbColorPrimary css-eg0mwd-MuiSlider-thumb"
          data-focusvisible="false"
          data-index="0"
          style="left: 0%; pointer-events: none;"
        >
          <input
            aria-orientation="horizontal"
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuenow="0"
            data-index="0"
            max="100"
            min="0"
            step="1"
            style="border: 0px; height: 100%; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 100%; direction: ltr;"
            type="range"
            value="0"
          />
          <span
            aria-hidden="true"
            class="MuiSlider-valueLabel css-nnid7-MuiSlider-valueLabel"
          >
            <span
              class="MuiSlider-valueLabelCircle"
            >
              <span
                class="MuiSlider-valueLabelLabel"
              >
                0
              </span>
            </span>
          </span>
        </span>
      </span>
      <div
        class="MuiGrid-root MuiGrid-container css-fvskug-MuiGrid-root"
      >
        0
      </div>
    </div>
  </div>
</DocumentFragment>
`;
